### PR TITLE
Automatically build and publish cross-platform binaries

### LINKED ISSUE #11: Pre-build and publish binaries as release
**Problem**
As a developer I want to run `clap-info` on a CLAP plugin to extract metadata. Ideally during an automated pipeline or script.
However due to the current setup, every pipeline/script execution has to run the steps:
```
git clone https://github.com/surge-synthesizer/clap-info
cd clap-info
git submodule update --init --recursive
cmake -Bbuild
cmake --build build
```
This requires downloading the git repo and git submodules dependencies (16MB), ensuring the correct `cmake` is installed, running `cmake` and then locating the built binary, which could be in different places per platform:
```
./build/clap-info
./build/clap-info
./build/Release/clap-info.exe
```

**Solution**
Pre-build the binaries once, and publish versions as GitHub releases e.g.:
* clap-info-linux.zip (289 KB)
* clap-info-mac.zip (377 KB)
* clap-info-win.zip (158 KB)

Users can then simply download the binary they need:
```
curl https://github.com/surge-synthesizer/clap-info/releases/download/v1.0.0/clap-info-linux.zip
unzip clap-info-linux.zip
./clap-info-linux/clap-info "/path/to/Surge XT.clap"
```

Ideally these would also be published to system package registries to enable an even faster installation:
```
apt install clap-info
brew install clap-info
choco install clap-info
```

### PR DESCRIPTION
**Pull request by kmturley**
Fixes #11 

GitHub Action which is triggered whenever you create a git tag:
`git tag v1.0.0`

It will automatically build and publish binaries for Linux, Mac and Windows

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        run: choco install zip
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Build binary
+        run: |
+          cmake -S . -B ./build
```
Comment (baconpaul) [inline .github/workflows/release.yml:59]: Doesn’t this need a cmake build type of release at the first cmake for non multi config generators?

Comment (kmturley) [reply .github/workflows/release.yml:59]: Ah you're correct:
https://cmake.org/cmake/help/latest/guide/tutorial/Packaging%20Debug%20and%20Release.html

will add

**Comment by kmturley** [comment]
Now it will run on git tag or PR
```
on:
  pull_request:
    branches:
      - main
  push:
    tags:
      - "v*"
```

File: .github/workflows/release.yml (hunk 1 above)
Comment (baconpaul) [reply .github/workflows/release.yml]: Great. Oh and on osx we will need the architecture flags also. (I would happily do this but am on vaca without a laptop right now).

-DCMAKE_OSX_ARCHITECTURES=“arm64;x86_64"

Comment (baconpaul) [reply .github/workflows/release.yml]: If you ca. just add that I’m happy to merge from my phone and see what happens :)

and thanks again!

Comment (kmturley) [reply .github/workflows/release.yml]: done!

Comment (baconpaul) [reply .github/workflows/release.yml]: Ok cool will merge!

**Comment by baconpaul** [comment]
Oh I merged this just now but how does the pr triggered action avoid uploading the release? Thanks!

**Comment by kmturley** [comment]
This workflow will run when a pull request is made to main OR a git tag is pushed.

I wonder if you want a slightly different workflow where pull request creates a build artifact you can use for testing but not a release?

**Comment by baconpaul** [comment]
For a pull request why don’t we for now just not store the binary but make sure it runs with -h? I know how to do that with azure but not with actions but sure there must be a way. 

Also again - thanks for the help here! I’m back from travels in a week so we can def have this binary up and published and the Json fixed before mid July.

**Comment by kmturley** [comment]
Ah I think I've fixed it. I separate the workflows into two separate files:

**feature.yml**
Runs when a pull request is opened/updated. installs, builds and tests binaries and stores an artifacts in the pipeline if you want to test them locally
Screen Shot 2022-07-03 at 8 41 07 PM

**release.yml**
Runs when a git tag is pushed (an official release), installs, builds and tests binaries and uploads the to a GitHub release for users to download and install on their own machines.
Screen Shot 2022-07-03 at 8 44 08 PM

Have reopened the PR!

**Pull request merged**